packages/neuron-ui/src/containers/Main/index.tsx

```diff
@@ -147,14 +149,20 @@ const MainContent = () => {
   const history = useHistory()
   const location = useLocation()
   const {
-    app: { isAllowedToFetchList = true },
+    app: {
+      isAllowedToFetchList = true,
+      countDown,
```

Comment:
Is `countDown` necessary to be global?

Reply:
yep, it will be used to dispatch a request in another place.

Reply:
The `fee rate` can be fetched on demand, then `count down` starts. And `count down` ends on leaving the page.

Reply:
1、The countdown is associated with the interface call. They are synchronized, and now the interface call is on the main page, so the countdown accordingly is there. And even if the countdown is not displayed, the interface needs to be called periodically on other pages to update `fee_rate_statistics`.
2、The display of the countdown is on the send page. I use a global variable to pass this value, not context.

Reply:
> 



> Maybe you can wrapper get the fee rate as a hook, like this, then every want get `feeRate` and `countDown` by call this hook
> 
> ```ts
> const useGetFeeRate = () => {
>     const [countDown, setCountDown] = useState(30)
>     const [feeRate, setFeeRate] = useState()
>     useEffect(() => {
>         const timeout = setTimeout(() => {
>           setCountDown(v => v - 1)
>         }, 1000)
>         return () => clearTimeout(timeout)
>     }, [])
>     useEffect(() => {
>      if (countDown === 0) {
>         getFeeRateStats()
>               .then(res => {
>                 const { mean, median } = res
>                 const suggested = res ? Math.max(1000, Number(mean), Number(median)) : 0
>                 setFeeRate({ ...res, suggestFeeRate: suggested })
>               })
>               .finaly {
>                   setCountDown(30)
>                }
>         }
>      }
>     }, [countDown])
>     return { feeRate, countDown }
> }
> ```

Thx.  I submitted a new version with a hook before I saw your response. Thanks for your help again.